demo05_basic_defender.py

```diff
@@ -14,10 +14,9 @@
 
 
 def move(bot, state):
-
-    if state is None:
+    if 'walls' not in state:
         # initialize the state object to be a graph representation of the maze
-        state = walls_to_graph(bot.walls)
+        state['walls'] = walls_to_graph(bot.walls)
```

Comment:
This should not be called `walls`. It is actually a graph of the non-walls positions in the maze, so a better name is needed. Incidentally, the `walls_to_graph` function needs to be renamed accordingly, maybe `maze_to_graph`?

Reply:
corridors? paths? non_walls?

Reply:
allowed?

Reply:
why not simply `graph`? If the corresponding function gets the right name, then it is unambiguous...

Reply:
I changed this to "allowed". It seems to work nicely, 'cause:
`networkx.shortest_path(state['allowed'], bot.position, target)[1]` is imho quite natural.

Reply:
This deserves a separate issue, but: Why don’t we remove walls altogether and only have the free positions? Coding with corridors makes more sense than coding with walls, no? (Although speaking about it seems to be easier with walls.) This could also lead to a speedup when we check that some move is legal.

Reply:
Well, calling it _free_ positions may be unclear, because if a bot or a food pellet is in that position, then that square is not _free_. Anyway, in pelita we could of course have _both_ sets, walls and free, so that everyone can choose their more intuitive coding style.

Reply:
Yeah, *free* is not good, but more because I think it is a bad name in general. I don’t think people will find it confusing; the concept of food is rather orthogonal to ‘freeness’ in my thinking.
-0 on having to ways to do it. I think *that* might be confusing.

My take: `legal_positions` for *all* allowed positions and `legal_moves` for the allowed neighbouring positions. 🙃

I will try to find some precedences for the naming. We can still resolve this at a later point. The distinction legal–allowed is not the best one either, I believe.

Reply:
It's called 'graph' in other places. I'll rename it here to match. If we want to change to something more descriptive, it should be done in all places at once.